granulate_utils/linux/ns.py

```diff
@@ -282,3 +282,18 @@ def resolve_host_path(process: Process, ns_path: str) -> str:
     Get a path in the host mount namespace pointing to path in process mount namespace.
     """
     return resolve_proc_root_links(get_proc_root_path(process), ns_path)
+
+
+def get_host_pid(nspid: int, container_id: str) -> Optional[int]:
+    assert len(container_id) == 64, f"Invalid container id {container_id!r}"
+    pid_namespace = ""
+    for process in process_iter():
+        try:
+            if not pid_namespace and container_id in Path(f"/proc/{process.pid}/cgroup").read_text():
```

Comment:
Here we should be using `_get_process_container_id` from gProfiler, I'll move to this repo and fix the code here; meanwhile we can merge this one.

Reply:
Fixed in https://github.com/Granulate/granulate-utils/pull/25/files